tcl/pdtk_canvas.tcl

```diff
@@ -508,15 +508,34 @@ proc ::pdtk_canvas::cleanname {name} {
     return $name
 }
 
-proc ::pdtk_canvas::cords_to_foreground {mytoplevel {state 1}} {
+proc ::pdtk_canvas::adjust_color {hex} {
+    # Parse hex color string "#RRGGBB"
+    scan $hex "#%2x%2x%2x" r g b
+    
+    foreach val {r g b} {
+        set v [set $val]
+        if {$v < 128} {
+            # If dark channel, brighten by adding 31 (clamped to 255)
+            set v [expr {min(255, $v + 0x1F)}]
+        } else {
+            # If light channel, darken by subtracting 31 (clamped to 0)
+            set v [expr {max(0, $v - 0x1F)}]
+        }
+        set $val $v
+    }
+    
+    # Format back to hex string
+    format "#%02x%02x%02x" $r $g $b
+}
+
+proc ::pdtk_canvas::cords_to_foreground {mytoplevel {state 1} fg bg sel} {
```

Comment:
hmm, i think that when doing such calculations on the GUI side (besides the question whether this is a good idea or not), then the GUI should have an a-priori knowledge of the colours involved.

that is: do **not** send the fg/bg/sel colors along with the `cords_to_foreground` command, but instead tell the GUI about the default colors when they are set (that is, in `glob_colors`).


---
also, i think that the proc signature `{mytoplevel {state 1} fg bg sel}` does not make any sense (the `state` variable loses becomes required).
this of course is not a problem if the additional arguments go away again :-)

Reply:
> how do I tell the GUI about the default colors when they are set?

I wonder if this means the GUI can then also change the Pd Console window. 

Any guidance in how to do that is highly appreciated. I'd like to try and do that here.

Also, we still have the issue with the edit text blinking bar, which is always black. This PR could expand and fix all these issues

Reply:
> I didn't really understand this... 

yea sorrty, i think my mobile's autocompletion got in the way again. i've corrected my original sentence and it should be more clear now.


> but I have to say I just tried something that seemed to make sense to me and it worked

i'll take this as: "I'm just experimenting a bit with ideas, and see what comes out of it".
as a consequence, i don't think this PR is ready for inclusion in the pending 0.56 release

Reply:
@umlaeute at least https://github.com/pure-data/pure-data/pull/2681/commits/64d2152f25b5c02c908f89bffae5105c074210f4 is a simple fix that should go into develop.

As for the rest, I have to sign off for the rest of the month. I wish I could take care of more stuff but I'm just really busy now. 

I think that fixing the highlight bug is important for 0.56 and if I can't nail if for now, I hope someone else gets to make a proper fix. One way or another, I think the current solution is good enough as a fix, at least in my tests... and things can be improved later.

Reply:
>  i've corrected my original sentence and it should be more clear now.

Cool, but I still don't why the state variable becomes required, how it wasn't before and why this is bad. One way or another, the main issue is: how do we tell the GUI about the default colors when they are set?